include/openenclave/internal/ecall_context.h

```diff
@@ -16,6 +16,11 @@ typedef struct _oe_ecall_context
     oe_call_host_function_args_t ocall_args;
     uint64_t ocall_buffer_size;
     uint8_t* ocall_buffer;
+
+    // Exit frame information for ocall stack stitching.
+    uint64_t debug_eexit_rip;
```

Comment:
This looks like SGX specific fields in a non-SGX-specific file.  If these are SGX-specific, please move them to an SGX-specific file.   For example, define a structure with these three fields and call it (say) oe_ecall_sgx_context_t, and then have a union here with that structure as one of the members.

Reply:
> 
> 
> This looks like SGX specific fields in a non-SGX-specific file. If these are SGX-specific, please move them to an SGX-specific file. For example, define a structure with these three fields and call it (say) oe_ecall_sgx_context_t, and then have a union here with that structure as one of the members.

Ah ok. I didn't realize that we had subdivisions (sgx/optee) under internal.
Entire ecall_context.h can be moved to internal/sgx

Reply:
I have moved ecall_context.h to internal/sgx/ecall_context.h.
Also created a task #2594 that is a good first task for a new contributor.